### PR TITLE
Clean up test output

### PR DESCRIPTION
**Pull request by mdickinson**
Silence warnings and remove a `print` in a test.

### CODE REVIEW

File: traits/etsconfig/tests/test_etsconfig.py [hunk 1]
```diff
@@ -258,7 +256,7 @@ def test_provisional_toolkit(self):
 
         with mock_sys_argv(test_args):
             with mock_os_environ(test_environ):
-                print(repr(self.ETSConfig.toolkit))
+                repr(self.ETSConfig.toolkit)
```
Comment (rahulporuri) [inline traits/etsconfig/tests/test_etsconfig.py]: why remove the print but keep the call to `repr`?

Comment (mdickinson) [reply traits/etsconfig/tests/test_etsconfig.py]: To keep the effect of the test. It's (indirectly) testing that the `repr` doesn't fail, so I wanted to keep that test.

File: traits/tests/test_interface_checker.py [hunk 2]
```diff
@@ -392,7 +393,7 @@ class Foo(HasTraits):
         class Bar(HasTraits):
             foo = Instance(IFoo)
 
-        b = Bar(foo=Foo())
+        Bar(foo=Foo())
```
Comment (rahulporuri) [inline traits/tests/test_interface_checker.py]: im guessing this too is a test where we simply instantiate the object and expect to see/not see errors.

File: traits/tests/test_category.py [hunk 3]
```diff
@@ -89,15 +89,15 @@ def test_subclass_extension_category(self):
...
+            self.base.pp
             self.fail(
                 msg="base.pp should have thrown AttributeError "
                 "as Category subclassing is not supported."
             )
         except AttributeError:
             pass
 
-        basepp = BasePlusPlus()
+        BasePlusPlus()
```
Comment (rahulporuri) [inline traits/tests/test_category.py]: im guessing that this too is a test where we instantiate the object and we expect to see/not-see errors.

File: traits/tests/test_interface_checker.py (hunk 2 above)
Comment (mdickinson) [reply traits/tests/test_interface_checker.py]: Yes, this was a drive-by fix to eliminate a `flake8` warning about an unused variable, without changing the semantics of the test.

File: traits/tests/test_category.py (hunk 3 above)
Comment (mdickinson) [reply traits/tests/test_category.py]: Yep.

**Pull request merged**